airflow/models/taskinstance.py

```diff
@@ -719,6 +719,7 @@ def current_state(self, session: Session = NEW_SESSION) -> str:
             .filter(
                 TaskInstance.dag_id == self.dag_id,
                 TaskInstance.task_id == self.task_id,
+                TaskInstance.map_index == self.map_index,
```

Comment:
It’s probably a good chance to rewrite this to something like

```python
from sqlalchemy.inspection import inspect

session.query(TaskInstance.state).filter(
    col == getattr(self, col.name)
    for col in inspect(TaskInstance).primary_key
).scalar()
```

This would be resilient to any primary key changes in the future.

Reply:
Thanks!

I've looked at the tests in tests/model.py and I don't see any examples of a test of a mapped task there. Is there any test that you would suggest so that a regression does not occur in the future?

Reply:
Honestly `current_state` isn’t really used almost anywhere in the code base (the only is for `airflow tasks state`, but I’d challenge even that usage is not necessary at all), so the test coverage is mostly non-existent. You can probably add a test for the `airflow tasks` CLI command (in `tests/cli/commands/test_task_command.py`) to cover this.

Reply:
Thanks, will work on this PR tomorrow and hopefully get it ready for review shortly